go/enclave/crosschain/block_message_extractor.go

```diff
@@ -44,13 +44,6 @@ func (m *blockMessageExtractor) Enabled() bool {
 func (m *blockMessageExtractor) StoreCrossChainMessages(block *common.L1Block, receipts common.L1Receipts) error {
 	defer m.logger.Info("Block cross chain messages processed", log.BlockHashKey, block.Hash(), log.DurationKey, measure.NewStopwatch())
 
-	areReceiptsValid := common.VerifyReceiptHash(block, receipts)
```

Comment:
We were verifying the receipt hash by reconstructing it using all the receipts but that check is not going to be feasible.

Reply:
how can we do it? Using Merkle branches?

Reply:
LLM has convinced me we could indeed provide a merkle proof for each receipt and that could give it high confidence the receipts are legit (so long as it can trust the receipt root hash).

Doesn't seem that hard to do but prefer to leave it out of this PR. Shall I leave a todo and create a GH issue?

Reply:
Actually would we be able to do it without having to request all the receipt data... I guess not. So maybe we need to resign ourselves to doing a receipt lookup for every transaction anyway? We need to have a think about this.

I'm going to raise another PR that caches the receipts because part of the data problem is re-lookups when things are re-submitted I think.